operator/v1/types_machineconfiguration.go

```diff
@@ -65,8 +74,58 @@ type MachineConfigurationStatus struct {
 	// +optional
 	NodeDisruptionPolicyStatus NodeDisruptionPolicyStatus `json:"nodeDisruptionPolicyStatus"`
+
+	// nodeSynchronizersStatus is the status of the machines managed by the node synchronizers.
+	// +openshift:enable:FeatureGate=PinnedImages
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	NodeSynchronizersStatus []NodeSynchronizerStatus `json:"nodeSynchronizersStatus"`
 }
 
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.updatedMachineCount", message="machineCount must be greater than or equal to updatedMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.availableMachineCount", message="machineCount must be greater than or equal to availableMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.unavailableMachineCount", message="machineCount must be greater than or equal to unavailableMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.readyMachineCount", message="machineCount must be greater than or equal to readyMachineCount"
+type NodeSynchronizerStatus struct {
+	// name is the name of the node synchronizer.
+	// +kubebuilder:validation:Required
+	Name NodeSynchronizerType `json:"name"`
+	// machineCount is the number of machines that are managed by the node synchronizer.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	MachineCount int64 `json:"machineCount"`
+	// updatedMachineCount is the number of machines that have been updated by the node synchronizer.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	UpdatedMachineCount int64 `json:"updatedMachineCount"`
+	// readyMachineCount is the number of machines managed by the node synchronizer that are in a ready state.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	ReadyMachineCount int64 `json:"readyMachineCount"`
+	// availableMachineCount is the number of machines managed by the node synchronizer which are available.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	AvailableMachineCount int64 `json:"availableMachineCount"`
+	// unavailableMachineCount is the number of machines managed by the node synchronizer but are unavailable.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=0
+	UnavailableMachineCount int64 `json:"unavailableMachineCount"`
+	// observedGeneration is the last generation change that has been applied.
+	// +kubebuilder:validation:Minimum=0
+	// +optional
+	ObservedGeneration int64 `json:"observedGeneration,omitempty"`
+}
+
+// NodeSynchronizerType is a string enum used in a NodeSynchronizers object. They describe the type of node synchronizer.
+// +kubebuilder:validation:Enum:="PinnedImageSets"
+type NodeSynchronizerType string
+
+const (
+	// PinnedImageSets represents a node synchronizer for pinned image sets.
+	PinnedImageSets NodeSynchronizerType = "PinnedImageSets"
```

Comment:
I'm flexible on this aspect.  Named or free-form.  One option would be to have an instance in the slice for each pinnedimageset for example.  I'm happy to start here and potentially relax before going accessible-by-default.